dir.c

```diff
@@ -3471,12 +3475,13 @@ file_s_fnmatch(int argc, VALUE *argv, VALUE obj)
 }
 
 /*
- *  call-seq:
- *    Dir.home()       -> "/home/me"
- *    Dir.home("root") -> "/root"
+ * call-seq:
+ *   Dir.home(user_name = ENV['USERNAME'] || ENV['LOGNAME']) -> dirpath
```

Comment:
It's not exactly accurate to say that it uses `ENV['USERNAME'] || ENV['LOGNAME']`. It fetches for the current user, rather than reading from the environment variables.

For example, I can overwrite these environment variables without affecting the behaviour.

```ruby
puts Dir.home # => "/Users/peter"
ENV['USERNAME'] = ENV['LOGNAME'] = 'blah'
puts Dir.home # => "/Users/peter"
```

It might be more accurate to give this call-seq and then explain that it defaults to the current user when `user_name` is `nil`.

```suggestion
 *   Dir.home(user_name = nil) -> dirpath
```